src/GraphQL/DI/IConfigureExecution.cs

```diff
@@ -0,0 +1,33 @@
+using System;
+
+namespace GraphQL.DI
+{
+    /// <summary>
+    /// Allows configuration of execution options immediately prior to executing a document.
+    /// </summary>
+    public interface IConfigureExecution
```

Comment:
Did you have a real need for this or did you do based on general considerations?

Reply:
`IConfigureExecution` -> `IConfigureExecutionOptions`

Reply:
> Did you have a real need for this or did you do based on general considerations?

Absolutely.  Read the rest of the PR.  It's used by GraphQLBuilderBase, AddComplexityAnalyzer, AddDocumentListener, etc

Reply:
And what about rename IConfigureExecution to IConfigureExecutionOptions ?

Reply:
> And what about rename IConfigureExecution to IConfigureExecutionOptions ?

Makes sense to me.